packages/debug-test/CDPDebugger.ts

```diff
@@ -0,0 +1,560 @@
+   *
+   * Sends a Debugger.disable CDP message (see https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#method-disable).
+   * @returns {Promise<void>} Promise indicating that the debugger backend has received the command.
+   */
+  public async debuggerDisable(): Promise<void> {
+    const commandId = this.nextCommandId++;
+
+    // the actual CDP message
+    const command = JSON.stringify({
+      id: commandId,
+      method: 'Debugger.disable',
+      params: {},
+    });
+
+    // a short description for logging
+    const commandDescription = `Debugger.disable (id: ${commandId})`;
+
+    const expectedResponse = {id: commandId, result: {}};
+    const resultPromise = new Promise<void>((resolve, _reject) => {
+      this.expectedResponses.push([
+        `result response for ${commandDescription}`,
+        actualResponse => {
+          // relying on object comparison to be independent of key order differences
+          if (lodash.isEqual(expectedResponse, actualResponse)) {
+            resolve();
+            return true;
+          }
+          return false;
+        },
+      ]);
+    });
+
+    await this.wsOpened;
+    testLog.message(`sending ${commandDescription}`);
+    this.ws.send(command);
+    this.lastCommand = commandDescription;
+    return resultPromise;
+  }
+
+  /**
+   * Stops JavaScript execution on the next statement.
+   *
+   * Sends a Debugger.pause CDP message (see https://chromedevtools.github.io/devtools-protocol/tot/Debugger/#method-pause).
+   * @returns {Promise<void>} Promise indicating that the debugger backend has received the command.
+   * @fires CDPDebugger#paused The 'paused' event indicates that the debugger backend has reached the 'paused' state.
+   */
+  public async debuggerPause(): Promise<void> {
+    const commandId = this.nextCommandId++;
+
+    // the actual CDP message
+    const command = JSON.stringify({
+      id: commandId,
+      method: 'Debugger.pause',
+      params: {},
+    });
+
+    // a short description for logging
+    const commandDescription = `Debugger.pause (id: ${commandId})`;
+
+    const resultPromise = new Promise<void>((resolve, _reject) => {
+      const resultResponse = {id: commandId, result: {}};
+      this.expectedResponses.push([
+        `result response for ${commandDescription}`,
+        response => {
+          // relying on object comparison to be independent of key order differences
+          if (lodash.isEqual(response, resultResponse)) {
+            resolve();
+            return true;
+          }
+          return false;
+        },
+      ]);
+    });
+
+    this.expectedResponses.push([
+      `Debugger.paused response for ${commandDescription}`,
+      response => {
+        if (
+          response.hasOwnProperty('method') &&
+          response.method === 'Debugger.paused'
+        ) {
+          this.raise('paused', () => response.params);
+          return true;
+        }
+        return false;
+      },
+    ]);
+
```

Comment:
Im not quite following the subtle difference between how the command acknowledgement response and the event response are expected here .. The acknowledgement seems to be added to the expected set when the test waits for the test method (debuggerPause) but the event is added synchronously .. Could you clarify ? and even better comment ..